src/server/game/Entities/Creature/GossipDef.cpp

```diff
@@ -224,10 +224,13 @@ void PlayerMenu::SendGossipMenu(uint32 titleTextId, ObjectGuid objectGUID)
         {
             ++count;
             data << uint32(questID);
-            data << uint32(item.QuestIcon);
+            data << uint32(quest->IsAutoComplete() ? 0 : item.QuestIcon);
             data << int32(quest->GetQuestLevel());
             data << uint32(quest->GetFlags());              // 3.3.3 quest flags
-            data << uint8(0);                               // 3.3.3 changes icon: blue question or yellow exclamation
+            uint8 changeIcon = 0;
+            if (quest->IsAutoComplete() && quest->IsRepeatable())
+                changeIcon = 1;
+            data << uint8(changeIcon);                      // 3.3.3 changes icon: blue question or yellow exclamation
```

Comment:
Partial code from 3.3.5 GossipFrame.lua 
```
GossipFrameAvailableQuestsUpdate()
...
		if ( isDaily ) then
			titleButtonIcon:SetTexture("Interface\\GossipFrame\\DailyQuestIcon"); -- blue !
		elseif ( isRepeatable ) then
			titleButtonIcon:SetTexture("Interface\\GossipFrame\\DailyActiveQuestIcon"); -- blue ?
		else
			titleButtonIcon:SetTexture("Interface\\GossipFrame\\AvailableQuestIcon"); -- yellow !
		end
```
isDaily = QuestFlags & 0x1000
isRepeatable = this field

Reply:
Active quests (QuestIcon == 3 || QuestIcon == 4) only get 
```
GossipFrameActiveQuestsUpdate()
...
		if ( select(i+3, ...) ) then -- is completed
			titleButtonIcon:SetTexture("Interface\\GossipFrame\\ActiveQuestIcon");
		else
			titleButtonIcon:SetTexture("Interface\\GossipFrame\\IncompleteQuestIcon");
		end		
```